subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -212,12 +223,27 @@ private SnapshotHierarchy withWatcherChangeErrorHandling(SnapshotHierarchy curre
             // No stacktrace here, since this is a known shortcoming of our implementation
             LOGGER.warn("Watching not supported, not tracking changes between builds: {}", ex.getMessage());
             return stopWatchingAndInvalidateHierarchy(currentRoot);
+        } catch (InsufficientResourcesForWatchingException ex) {
+            logInsufficientResourcesException(ex);
+            return stopWatchingAndInvalidateHierarchy(currentRoot);
         } catch (Exception ex) {
             LOGGER.error("Couldn't update watches, not watching anymore", ex);
             return stopWatchingAndInvalidateHierarchy(currentRoot);
         }
     }
 
+    private void logInsufficientResourcesException(InsufficientResourcesForWatchingException ex) {
+        Optional<IncreaseLimitsDocumentation> documentation = insufficientResourcesForWatchingDocumentationIndex.getDocumentationFor(ex);
+        documentation.ifPresent(doc -> LOGGER.warn(
+            "Unable to watch the file system for changes, {}. See {} for more details.",
```

Comment:
The message here should be similar to the other one used for general problems (`Couldn't create watch service, not tracking changes between builds`).

Reply:
I agree that the messages should be similar. What do you think is better? Having messages like this:
- `Unable to watch the file system for changes, inotify watches limit too low. See ... for more details.`
- 'Unable to watch the file system for changes, couldn't create watch service`

Or:
- `Inotify watches limit too low, not tracking changes between builds. See ... for more details.`
- `Couldn't create watch service, not tracking changes between builds`

I think I prefer the first variant.

Reply:
First one sounds fine to me.

Reply:
I unified the messages.